app/models/pipeline_run.rb

```diff
@@ -845,6 +845,7 @@ def update_job_status
       # all stages succeeded
       self.finalized = 1
       self.job_status = STATUS_CHECKED
+      precache_report_info!
```

Comment:
Could you do this asynchronously using resque? I'm weary of adding more processing steps to `update_job_status`. It needs to be fast -- the round-robin of pipeline checks should not be delayed by the precaching of the reports

Reply:
Also, logically the precache action should probably be associated with the "result monitor" (and the results_finalized attribute of pipeline_run) rather than the "job monitor" (and the finalized attribute of pipeline_run). See https://github.com/chanzuckerberg/idseq-web/blob/9c55396980692f860c4bc961372f355c75017b16/app/models/pipeline_run.rb#L763 for the result-monitor equivalent of the `update_job_status` function.

Reply:
Thanks for the guidance. I wanted to ask about that. I'll move `precache_report_info!` to inside `monitor_results`.

Reply:
Do you still think this should be done non-blocking thru resque? My assumption is that the precaching happens only once per pipeline run success, which seems relatively rare.

Reply:
Yes, I still think if we want to open ourselves up to the possibility of precaching with all the different backgrounds or other parameters, it would be best to make it non-blocking

Reply:
I agree with Charles, some of these reports might be slow to generate, and it can affect the stability of update_status.

Reply:
Ok I'll do that in this PR.